include/zenoh-pico/session/session.h

```diff
@@ -130,7 +130,7 @@ _Z_ELEM_DEFINE(_z_questionable_sptr, _z_questionable_sptr_t, _z_noop_size, _z_qu
 _Z_LIST_DEFINE(_z_questionable_sptr, _z_questionable_sptr_t)
 
 typedef struct {
-    _z_reply_t _reply;
+    _z_reply_t *_reply;
```

Comment:
Just to confirm, you had to change it to a pointer in the heap in order to allow the user-code to get the ownership of the reply without an extra copy. 
Can you confirm if this was the case?

Reply:
Yes, that's the goal. _z_reply_t as a part of larger structure can't change ownership

Reply:
Smart pointer definition: https://github.com/eclipse-zenoh/zenoh-pico/blob/master/include/zenoh-pico/collections/pointer.h
Example of usage: https://github.com/eclipse-zenoh/zenoh-pico/search?q=sptr